packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -482,19 +477,38 @@ export const splitFormatIntoSections = <TDate>(
     return null;
   };
 
-  const splitFormat = (token: string) => {
-    const escapedParts = getEscapedPartsFromFormat(utils, token);
+  const splitFormat = (inputFormat: string) => {
```

Comment:
Since we no longer have recursion, we should be able to inline this function :+1: